modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -4683,3 +4688,150 @@ def compare(self, other, **kwargs):
                 other._modin_frame,
             )
         )
+
+    def case_when(self, caselist):
+        if Engine.get() in ("Unidist", "Dask"):
```

Comment:
Does Dask also fail?

Reply:
Yes.

Reply:
What error occurs? (For Dask)

Reply:
I once had a problem where some tests were failing/hanging for Unidist and Dask specifically (not for Ray), when I accidentally passed a distributed object to a kernel, for example passing a low-level PandasDataframe, or an object that contains a PandasDataframe in it (LazyCategorical, ModinDtypes, ModinIndex). Could you double-check that it's not the case here?

I also had a similar problem when declaring kernel functions in a global score (like `_case_when()` in you RP), could you try moving the declaration inside the qc method and see if it helps?

Reply:
Fixed for Unidist and Dask. It was required to do `execution_wrapper.put()` for it each callable argument. Moreover, for Dask it's required for each partition, otherwise CancelledError is raised.